LibertyDSNP/example-client
# Retrieve Graph Data from DSNP

Purpose
---------------
In order to know who a user follows (or unfollows), we need to retrieve graph change event from DSNP batch data.


Solution
---------------
This PR retrieves all graph change events from DSNP batches and uses it to populate the graph in Redux. It then uses that to update the UI in various places like the connection list, and the user's feed.


Change summary
---------------
* Alter populate script to add follow information
* Update the SDK to incorporate a parquet issue storing graph change events
* Read graph change events in the SDK
* Alter the redux store to store follows and followers efficiently
* Modify PostLIst to use the new graph change events.
* Modify ConnectionList to use the new graph change events.
* Fix tests.
* Update package-lock.json to version 2 (hopefully for the last time).

Steps to Verify
----------------
1. Start up a clean, deployed chain and the static server.
2. Run the new populate script.
3. Start the app.
4. Confirm that there are posts in the "Feed" tab of the post list.
5. Confirm that you see connections when you click "followers" in the profile block.
6. Confirm that you see connections when you click "following" in the profile block.
7. Confirm that the posts in the feed are consistent with the following list.
8. Confirm that the the follow/unfollow buttons in the "followers" section are consistent with the follow list.
